src/cookbooks/bigquery/optimization.md

```diff
@@ -1,28 +1,165 @@
-        - `Cannot query over table 'moz-fx-data-shared-prod.telemetry_derived.main_summary_v4' without a filter over column(s) 'submission_date' that can be used for partition elimination`
-    - See [_Writing Queries_](./querying.md#writing-queries) for examples.
-- Reduce data before using a JOIN
-    - Trim the data as early in the query as possible, before the query performs a JOIN. If you reduce data early in the processing cycle, shuffling and other complex operations only execute on the data that you need.
-    - Use sub queries with filters or intermediate tables or views as a way of decreasing sides of a join, prior to the join itself.
+# Optimizing BigQuery Queries
+
+When writing a query using [STMO](https://sql.telemetry.mozilla.org) or the BigQuery console, you can improve performance and reduce costs by learning how data is stored, how databases function, and what you can change about a query to take advantage of the storage structure and the data function.
+
+[Queries are charged by data scanned at \$5 per terabyte (TB)](https://cloud.google.com/bigquery/pricing#on_demand_pricing) so each 200 gigabytes of data scanned will cost \$1: on tables with hundreds of TBs of data (like the [main ping table](../../datasets/pings.md#main-ping) or [`clients_daily`](../../datasets/batch_view/clients_daily/reference.md)), costs **can add up very quickly**.
+
+## Table of Contents
+
+<!-- toc -->
+
+## TL;DR: What to implement for quick improvements
+
+- Filter on a partitioned column such as `submission_timestamp` or `submission_date` (_even_ if you have a `LIMIT`: see [optimization caveats](#optimization-caveats))
+- Use a sample of the data that is based on the `sample_id` field. This can be helpful for initial development even if you later run the query using the entire
+  population (without sampling).
+  - Tables that include a `sample_id` field will usually have that as one of the clustering fields and you can efficiently scan random samples of users by specifying `WHERE sample_id = 0` (1% sample), `WHERE sample_id < 10` (10% sample), etc. This can be especially helpful with `main_summary`, `clients_daily`, and `clients_last_seen` which are very large tables and are all clustered on `sample_id`.
+- Many datasets also cluster on `normalized_channel`, corresponding to the channel of the product. If you are working with data that has different channels (for example, Firefox desktop), limit your initial query to a channel with a limited population like Nightly (in the case of Firefox desktop, do this by adding `WHERE normalized_channel='nightly'` to your query)
+- Select only the columns that you want (**Don't** use `SELECT *`)
+  - If you are experimenting with data or exploring data, use one of the [data preview options](https://cloud.google.com/bigquery/docs/best-practices-costs#preview-data) instead of `SELECT *`.
+- Use approximate algorithms: e.g., `approx_distinct(...)` instead of `COUNT(DISTINCT ...)`
```

Comment:
```suggestion
- Use [approximate algorithms](https://cloud.google.com/bigquery/docs/reference/standard-sql/approximate_aggregate_functions): e.g., `approx_count_distinct(...)` instead of `COUNT(DISTINCT ...)`
```

Reply:
oops, see you have the link just below this

Reply:
I think I like this way better, less text. I'll apply this suggestion and delete the text below.

Reply:
I would change this from:

Many datasets also cluster on `normalized_channel`, corresponding to the channel of the product. If you are working with data that has different channels (for example, Firefox desktop), limit your initial query to a channel with a limited population like Nightly (in the case of Firefox desktop, do this by adding `WHERE 

To 

Many datasets also cluster on `normalized_channel` and correspond to the channel of the product. If you work with data that has been assigned to different channels (for example, Firefox desktop), limit your initial query to a channel that has a limited population, such as Nightly (in the case of Firefox desktop, adding `WHERE . . . .


Change this from:

If you are experimenting with data or exploring data, use . . . 

To:

If you experiment with or explore data, use . . .